src/modules/assignment/assignment.service.ts

```diff
@@ -14,16 +19,37 @@ export class AssignmentService {
 
   // fetch all Assignments
   async getAllAssignment(): Promise<Assignment[]> {
-    const Assignments = await this.AssignmentModel.find().exec();
-    return Assignments;
+    try {
+      const Assignments = await this.AssignmentModel.find().exec();
+      return Assignments;
+    } catch (e) {
+      throw new HttpException(
+        {
+          status: HttpStatus.INTERNAL_SERVER_ERROR,
+          error: `${e}`,
+        },
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
   }
 
   // Get a single Assignment
   async getAssignment(AssignmentId): Promise<Assignment> {
-    const Assignment = await this.AssignmentModel.findById(AssignmentId).exec();
-
-    if (Assignment) {
-      return Assignment;
+    try {
+      const Assignment = await this.AssignmentModel.findById(
+        AssignmentId,
+      ).exec();
+      if (Assignment) {
+        return Assignment;
+      }
+    } catch (e) {
+      throw new HttpException(
+        {
+          status: HttpStatus.INTERNAL_SERVER_ERROR,
+          error: `${e}`,
+        },
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
```

Comment:
See nest exception filter, It is not the nest way to throw internal server error

Reply:
https://docs.nestjs.com/exception-filters

in the nestjs docs this method was given

Reply:
Are u talking about the InternalServerErrorException() ?

Reply:
@Abhishek-kumar09 I've replaced it now with InternalServerErrorException() of nestJs